src/libpmemfile-posix/file.c

```diff
@@ -1252,14 +1247,32 @@ vinode_rename(PMEMfilepool *pfp,
 
 		struct pmemfile_time t;
 		file_get_time(&t);
-		vinode_add_dirent(pfp, dst_parent, new_name, new_name_len,
-				src_vinode, t);
 
-		vinode_unlink_file(pfp, src_parent, src_dirent, src_vinode);
+		if (src_parent == dst_parent) {
+			/* optimized rename */
+			pmemobj_tx_add_range_direct(src_dirent->name,
+					new_name_len + 1);
+
+			strncpy(src_dirent->name, new_name, new_name_len);
+			src_dirent->name[new_name_len] = '\0';
 
-		if (vinode_is_dir(src_vinode) && src_parent != dst_parent)
-			vinode_update_parent(pfp, src_vinode, src_parent,
-					dst_parent);
+			/*
+			 * From "stat" man page:
+			 * "st_mtime of a directory is changed by the creation
+			 * or deletion of files in that directory."
+			 */
+			TX_SET_DIRECT(src_parent->inode, mtime, t);
```

Comment:
No. mtime is modified in vinode_add_dirent and vinode_unlink_file called when src_parent != dst_parent.